operators/tokenizer/bpe_utils.hpp

```diff
@@ -0,0 +1,231 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include "ocos.h"
+#include "narrow.h"
+
+#include <algorithm>
+#include "ustring.h"
+
+#include "unicode.h"
+
+class SpecialTokenMap {
+ public:
+  void Add(ustring p_str, int p_id) {
+    auto it = token_map_.find(p_str);
+    if (it != token_map_.end()) {
+      if (it->second != p_id) {
+        ORTX_CXX_API_THROW("Duplicate special tokens.", ORT_INVALID_ARGUMENT);
+      }
+    } else {
+      token_map_[p_str] = p_id;
+      token_list_.push_back(SpecialTokenInfo(std::move(p_str), p_id));
+    }
+  }
+
+  std::list<std::pair<ustring, int>> SplitBySpecialTokens(ustring input) const {
```

Comment:
looks like the `std::list` only gets appended to. consider `std::vector` instead as it has less dynamic allocation overhead.

Reply:
general: usually `vector` is a better choice if you don't need to frequently insert or remove from locations other than the end.

Reply:
One concern about std::vector related to this case is when element is large how efficient std::vector can move them into the new buffer on growing vector buffer.